src/search/ir_sema_checker.h

```diff
@@ -50,6 +50,9 @@ struct SemaChecker {
         GET_OR_RET(Check(v->query_expr.get()));
         if (v->limit) GET_OR_RET(Check(v->limit.get()));
         if (v->sort_by) GET_OR_RET(Check(v->sort_by.get()));
+        if (v->sort_by && v->sort_by->IsVectorField() && !v->limit) {
```

Comment:
maybe here we also need to consider if the current field is a `noindex` field?

Reply:
Just curious, is a `NOINDEX` vector field the only type of index that completely prohibits all query operations, such as KNN and range queries? For other field types, like tag and numeric filters, queries simply revert to a full scan right?

Reply:
I think we can allow range queries for noindex vector fields. But for KNN we don't need to support it for noindex fields.